docs/upgrading.asciidoc

```diff
@@ -0,0 +1,18 @@
+[[upgrading]]
+== Upgrading
+Upgrades between minor versions of the agent, like from 1.1 to 1.2 are always backwards compatible.
+Upgrades that involve a major version bump often come with some backwards incompatible changes.
+
+Before upgrading the agent, be sure to review the:
+
+* <<release-notes,Agent release notes>>
+* {apm-overview-ref-v}/agent-server-compatibility.html[Agent and Server compatibility chart]
```

Comment:
regarding updating the agent, since the latest agent version is the recommended one:
what about providing a bash one-liner to download the latest version from maven central ?

Also, it might be relevant to add that the recommended way to update (at least for now) is the following:
- shut-down the application
- update agent jar
- (optionally) change JVM settings if path to agent jar has changed due to a different filename for example
- re-start the application.

One extra thing that we could add maybe later is to have a shell one-liner script to download latest version of the agent for easier update/install.

Reply:
Could be done like this: https://stackoverflow.com/a/26288401/1125055

Reply:
Thanks Sylvain. I've added in the upgrade steps. I'm having trouble adapting the one-liner that Felix provided to the APM jar